### DISCUSSION TITLE
Help with volume estimation error

### DISCUSSION THREAD

**Discussion by orefram** [marcomusy/vedo#1059] [Q&A] [answered]
Hi everyone, 

I need some help with a simple code that I made to estimate volume. I started with two intersecting cubes ( Dims = 10 x 10 x 10 ), and the only thing I did was move it 5 units on the y axis. 

```
from vedo import *

Cube_a = Box(pos = (10,10,10), size=(10, 10, 10)).triangulate()
Cube_b = Box(pos = (10,15,10), size=(10, 10, 10)).triangulate()
Intersection_ab=  Cube_a.boolean("intersect", Cube_b).c('blue')

print('Vol. Intersection = ', Intersection_ab.volume())
show(Intersection_ab, axes=1, viewup='z').close()
```

The output is this:
**Vol. Intersection =  642.8571428571427 // (should be 500)**

![image](https://github.com/marcomusy/vedo/assets/140839672/f0709310-0dad-4b43-a63b-963cb4bebd5a)

How can I solve this?
Thank you very much in advance

**Comment by marcomusy** [comment, marked as answer]
As you can see you have colliding faces! Allow a  small tolerance to avoid it:

```py
from vedo import *

cube_a = Box(pos = (10,10,10), size=(10, 10, 10)).triangulate().alpha(0.2)
cube_b = Box(pos = (10,15,10), size=(10, 10, 10)).triangulate().c("purple5")
cube_a.scale(1.001)  # make it a bit bigger to avoid colliding faces
intersection_ab=  cube_a.boolean("intersect", cube_b).c('blue5')

print('Vol. Intersection = ', intersection_ab.volume())
show([[cube_a, cube_b], intersection_ab], N=2, axes=1, viewup='z').close()
```

![Screenshot from 2024-02-27 15-10-34](https://github.com/marcomusy/vedo/assets/32848391/78753820-c035-4376-9f6c-39df8c2f1ff2)

```
Vol. Intersection =  500.500011444092
```
